unixFilter doesn't return stderr output on failure

Hi
This is something that comes up quite a lot for us...

### Steps
1. Use unixFilter, e.g.  as per docs
```
match "style.scss" $ do
    route   $ setExtension "css"
    compile $ getResourceString >>=
        withItemBody (unixFilter "sass" ["-s", "--scss"]) >>=
        return . fmap compressCss
```
2. Include a SASS error in your `style.scss` (e.g. a line of  `@@`).

### Expected
The error message from SASS, to help us fix the problem. 

### Actual
```  [ERROR] Hakyll.Core.UnixFilter.unixFilterWith: sass -s --scss  gave exit code 65.```
You can then run SASS locally (if you're building locally - we build it in Docker so this isn't so easy) to find the output:

```
Error: Invalid CSS after "@": expected identifier, was "@@ "
        on line 1 of standard input
  Use --trace for backtrace.
```

It would be great if Hakyll (a great library - thanks BTW) passed on this output

--- a/lib/Hakyll/Core/UnixFilter.hs
+++ b/lib/Hakyll/Core/UnixFilter.hs
@@ -92,7 +92,8 @@ unixFilterWith writer reader programName args input = do
         ExitSuccess   -> return output
         ExitFailure e -> fail $
             "Hakyll.Core.UnixFilter.unixFilterWith: " ++
-            unwords (programName : args) ++ " gave exit code " ++ show e
+            unwords (programName : args) ++ " gave exit code " ++ show e ++
+            ". (Error: " ++ err ++ ")"
 
 
 --------------------------------------------------------------------------------

--- a/tests/Hakyll/Core/UnixFilter/Tests.hs
+++ b/tests/Hakyll/Core/UnixFilter/Tests.hs
@@ -17,6 +17,7 @@ import           Hakyll.Core.Compiler
 import           Hakyll.Core.Compiler.Internal
 import           Hakyll.Core.Item
 import           Hakyll.Core.UnixFilter
+import           Hakyll.Core.Identifier
 import           TestSuite.Util
 
 
@@ -25,16 +26,19 @@ tests :: TestTree
 tests = testGroup "Hakyll.Core.UnixFilter.Tests"
     [ testCase "unixFilter rev"   unixFilterRev
     , testCase "unixFilter false" unixFilterFalse
+    , testCase "unixFilter error" unixFilterError
     ]
 
+testMarkdown :: Identifier
+testMarkdown = "russian.md"
 
 --------------------------------------------------------------------------------
 unixFilterRev :: H.Assertion
 unixFilterRev = do
     store    <- newTestStore
     provider <- newTestProvider store
-    output   <- testCompilerDone store provider "russian.md" compiler
-    expected <- testCompilerDone store provider "russian.md" getResourceString
+    output   <- testCompilerDone store provider testMarkdown compiler
+    expected <- testCompilerDone store provider testMarkdown getResourceString
     H.assert $ rev (itemBody expected) == lines (itemBody output)
     cleanTestEnv
   where
@@ -47,10 +51,24 @@ unixFilterFalse :: H.Assertion
 unixFilterFalse = do
     store    <- newTestStore
     provider <- newTestProvider store
-    result   <- testCompiler store provider "russian.md" compiler
+    result   <- testCompiler store provider testMarkdown compiler
     H.assert $ case result of
         CompilerError es -> any ("exit code" `isInfixOf`) es
         _                -> False
     cleanTestEnv
   where
     compiler = getResourceString >>= withItemBody (unixFilter "false" [])
+
+
+--------------------------------------------------------------------------------
+unixFilterError :: H.Assertion
+unixFilterError = do
+    store    <- newTestStore
+    provider <- newTestProvider store
+    result   <- testCompiler store provider testMarkdown compiler
+    H.assert $ case result of
+        CompilerError es -> any ("invalid option" `isInfixOf`) es
+        _                -> False
+    cleanTestEnv
+  where
+    compiler = getResourceString >>= withItemBody (unixFilter "head" ["-#"])

